config/config/src/isRcSetting.ts

```diff
@@ -0,0 +1,5 @@
+import npmTypes from '@pnpm/npm-conf/lib/types.js'
+import { types } from './types.js'
+
+export const isRcSetting = (kebabKey: string, extraTypes: Record<string, unknown> = {}): boolean =>
+  kebabKey.startsWith('@') || kebabKey.startsWith('//') || kebabKey in npmTypes || kebabKey in types || kebabKey in extraTypes
```

Comment:
I'd limit this to just these:
```suggestion
  kebabKey.startsWith('@') || kebabKey.startsWith('//')
```

+ any other auth related settings.